xsuite/src/proxy/proxy.ts

```diff
@@ -29,22 +29,38 @@ export class Proxy {
     this.blockNonce = params.blockNonce ?? null;
   }
 
-  fetchRaw(path: string, data?: any) {
-    const baseUrl = this.proxyUrl;
-    const init: RequestInit = { headers: this.headers };
-    const url = new URL(`${baseUrl}${path}`);
-    if (this.blockNonce !== null) {
-      url.searchParams.append("blockNonce", this.blockNonce.toString());
+  private makeUrl(
+    path: string | URL,
+    params: Record<string, { toString: () => string } | undefined | null> = {},
+    override: boolean = false,
+  ) {
+    const url = new URL(path, this.proxyUrl);
+    for (const [k, v] of Object.entries(params)) {
+      if (v !== undefined && v !== null) {
+        if (override) {
+          url.searchParams.set(k, v.toString());
+        } else {
+          url.searchParams.append(k, v.toString());
+        }
+      }
     }
+    return url;
+  }
+
+  fetchRaw(url: URL, data?: any) {
```

Comment:
Isn't this breaking? Can I still pass a path as before (e.g. `/address/[address]`)?

Furthermore, is there even a need for developers to have the type URL there?

Reply:
Additional note: I didn't catch it in the previous PR but it seems that if I call `fetchRaw`, it won't apply the block nonce on it? Not sure that this is the good default?

Reply:
Is there a need to have `fetchRaw` as a public method? If so, indeed, I'll have to make some changes. If not, I'll just make `fetchRaw` private?

Reply:
Yes, it's useful to have `fetchRaw` as a public method, because sometimes you want to really have no post-processing on the response (e.g. debugging).

Regarding the additional note, did I get it right?

As a general note, for the next PRs, if you envisage to introduce breaking changes, it is better if we discuss them while you implement. We can do some breaking changes but we should strive to minimize them as much as possible.